### PR TITLE
Client Encryption: Adds Client Encryption APIs to GA SDK

### PR DESCRIPTION
**Pull request by anujtoshniwal**
This reverts commit b73ed615e03aed0e7adec5ec309ad3eecc2d293a.

# Pull Request Template

## Description

Need this to GA the client-encryption feature.

## Type of change

Please delete options that are not relevant.

- [] Bug fix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [] This change requires a documentation update

## Closing issues

To automatically close an issue: closes #IssueNumber

### CODE REVIEW

**Review by abhijitpai — APPROVED** [review]

**Review by kr-santosh — APPROVED** [review]

**Review by FabianMeiswinkel — CHANGES_REQUESTED** [review]
We had arch board review with the Azure Central team yesterday for the Java version of the library. There was some feedback that needs further discussion with them and is relevant across Java and .Net (mostly how to expose way for crypto algorithms and how to wrap/unwrap keys) - please hold this PR until we have a clear way forward there (because it could impact public surface area)

**Comment by anujtoshniwal** [comment]
> 

Hi @FabianMeiswinkel, @j82w, @kirankumarkolli, now that the feedback has been addressed, please help review and close on this PR. Thanks!

**Review by abhijitpai — DISMISSED** [review]

**Review by kr-santosh — APPROVED** [review]

File: Microsoft.Azure.Cosmos/src/Resource/Database/DatabaseCore.cs [hunk 1]
```diff
@@ -1,909 +1,894 @@
...
-              trace: trace,
-              cancellationToken: cancellationToken);
-        }
-    }
-}
+﻿//------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.  All rights reserved.
+//------------------------------------------------------------
+
+namespace Microsoft.Azure.Cosmos
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/src/Resource/Database/DatabaseCore.cs:5]: Please revert the line ending changes.

**Review by kirankumarkolli — DISMISSED** [review]

**Review by FabianMeiswinkel — APPROVED** [review]
LGTM - thanks

**Review by j82w — APPROVED** [review]

**Pull request merged**